src/turn.c

```diff
@@ -112,13 +112,13 @@ static int find_ordered_transaction_id_rec(turn_entry_t *const ordered_transacti
                                            const uint8_t *transaction_id, int begin, int end) {
 	int d = end - begin;
 	if (d <= 0)
-		return begin;
+		return -1;
 
-	int pivot = d / 2;
+	int pivot = begin + d / 2;
```

Comment:
Good catch! Could you please also fix the binary search for the channel just before? It has the same mistake...

Reply:
done